cloud/pkg/cloudhub/channelq/channelq.go

```diff
@@ -108,19 +116,39 @@ func (q *ChannelMessageQueue) addMessageToQueue(nodeID string, msg *beehiveModel
 		// the version stored in the database
 		if !exist {
 			resourceNamespace, _ := edgemessagelayer.GetNamespace(*msg)
+			resourceName, _ := edgemessagelayer.GetResourceName(*msg)
 			resourceUID, err := GetMessageUID(*msg)
+			objectSyncName := synccontroller.BuildObjectSyncName(nodeID, resourceUID)
 			if err != nil {
 				klog.Errorf("fail to get message UID for message: %s", msg.Header.ID)
 				return
 			}
-			objectSync, err := q.objectSyncLister.ObjectSyncs(resourceNamespace).Get(synccontroller.BuildObjectSyncName(nodeID, resourceUID))
+			objectSync, err := q.objectSyncLister.ObjectSyncs(resourceNamespace).Get(objectSyncName)
 			if err == nil && objectSync.Status.ObjectResourceVersion != "" && synccontroller.CompareResourceVersion(msg.GetResourceVersion(), objectSync.Status.ObjectResourceVersion) <= 0 {
 				return
+			} else if err != nil && apierrors.IsNotFound(err) {
+				objectSync := &v1alpha1.ObjectSync{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: objectSyncName,
+					},
+					Spec: v1alpha1.ObjectSyncSpec{
+						ObjectAPIVersion: util.GetMessageAPIVerison(msg),
```

Comment:
Seems build failed because of this line :)

Reply:
Yes, the error says "GetMessageAPIVerison not declared by package util (typecheck)". But this line  is same as https://github.com/kubeedge/kubeedge/blob/master/cloud/pkg/cloudhub/handler/messagehandler.go#L559. And I can build and run it on my computer. I don't know why？

Reply:
Can you fetch  the latest master branch then rebase to it and then force push, let's see the result.

Reply:
Just have a check, `GetMessageAPIVerison` in which `Verison` is a typo, and fixed last month, you branch is based a old branch, and the ci will rebase your commit to latest master, so build will failed.

Reply:
yes, https://github.com/My-pleasure/kubeedge/blob/fix/reliable-message-delivery/pkg/metaserver/util/util.go#L121 need rebase to the upstream/master

Reply:
Ok, I know. But I may have made a mistack when I merge and rebase.... I'll try to fix it tomorrow.

Reply:
thanks a lot, we'd better use `git rebase` cmd to rebase to the head, but not using `git merge`, this will make commits too much.